src/Serilog.Sinks.Email/Sinks/Email/IEmailTransport.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright 2014 Serilog Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Serilog.Sinks.Email
+{
+    public interface IEmailTransport : IDisposable
```

Comment:
Does this have to be public?

Reply:
Hi, as said in the description, I had a `protected internal virtual` method in the `EmailConnectionInfo` class to create the concrete implementation of the `IEmailTransport`. This will allow two things:
1. anyone can subclassing the `EmailConnectionInfo` and provides its own implementation of the transport
2. in the tests, I can do the same thing and provide a mock implementation of the transport.

Anyway, I can "remove" it and made all of this purely internal.